query-engine/core/src/query_graph_builder/builder.rs

```diff
@@ -63,7 +63,9 @@ impl QueryGraphBuilder {
 
         let mut graph = match (&query_info.tag, query_info.model.clone()) {
             (QueryTag::FindUnique, Some(m)) => read::find_unique(parsed_field, m).map(Into::into),
+            (QueryTag::FindUniqueOrThrow, Some(m)) => read::find_unique_or_throw(parsed_field, m).map(Into::into),
```

Comment:
In order to build the new APIs I reused first/many/one modules in query_builder. I added a field of type `QueryOptions`to `RecordQuery` and `ManyRecordsQuery`. See [query_ast/read.rs](https://github.com/prisma/prisma-engines/pull/3356/files#diff-4fa00cec9f05ca95b632cdf050c031d5e3c4f95fb1a224dd5fdcda75199ff912)